state/protocol/badger/mutator.go

```diff
@@ -122,10 +120,12 @@ func NewFullConsensusState(
 //
 //	candidate.View == certifyingQC.View && candidate.ID() == certifyingQC.BlockID
 //
-// Caution:
-//   - This function expects that `certifyingQC` has been validated.
+// CAUTION:
+//   - This function expects that `qc` has been validated. (otherwise, the state will be corrupted)
 //   - The parent block must already be stored.
 //
+// Orphaned blocks are excepted.
```

Comment:
sorry, there was a typo. It should be "accepted". I have extended the goDoc to explain why this is needed: 
> Per convention, the protocol state requires that the candidate's parent has already been ingested.
Other than that, all valid extensions are accepted. Even if we have enough information to determine that
a candidate block is already orphaned (e.g. its view is below the latest finalized view), it is important
to accept it nevertheless to avoid spamming vulnerabilities. If a block is orphaned, consensus rules
guarantee that there exists only a limited number of descendants which cannot increase anymore. So there
is only a finite (generally small) amount of work to do accepting orphaned blocks and all their descendants.
However, if we were to drop orphaned blocks, e.g. block X of the orphaned fork X <- Y <- Z, we might not
have enough information to reject blocks Y, Z later if we receive them. We would re-request X, then
determine it is orphaned and drop it, attempt to ingest Y re-request the unknown parent X and repeat
potentially very often.